fibre/blob.go

```diff
@@ -52,17 +53,22 @@ type BlobConfig struct {
 	BlobVersion uint8
 	// CodingWorkers is the number of workers to use for encoding and decoding rsema1d.
 	CodingWorkers int
+	// ShardingFactor is the expected number of validators in the network.
+	// This is used to calculate the maximum gRPC message size per validator.
+	// Should be set to match the expected validator set size.
+	ShardingFactor int
```

Comment:
[nit] `ShardingFactor` is indirect and isn't as descriptive as something like `NumValidators` or `ValidatorCount`

```suggestion
	// ValidatorCount is the expected number of validators in the network.
	// This is used to calculate the maximum gRPC message size per validator.
	// Should be set to match the expected validator set size.
	ValidatorCount int
```

also this should probably say number of active validators because there are 269 validators on Mainnet but only 100 are in the active set. More specifically it could say the number of Fibre Service Providers.

Ref: https://www.mintscan.io/celestia/validators

Reply:
The higher-level constants can be descriptive on the protocol level. Ideally, there should be a canonical configuration, somehow like in `consts` that takes proper values named according to the core protocol. For the BlobConfig level, the validators are an external concept

Reply:
Needing to differentiate between active and non-active validators is also a reason not to bring in the notion of validators here. This parameter controls how many shards to split the blob into, not how many validators there is.

Reply:
`ShardingFactor` forces readers to learn a new term and understand a new relationship (`ShardingFactor` == number of Fibre service providers).

IMO using a new name for this adds cognitive complexity. We expect this parameter to  match the number of Fibre service providers so I still vote for a name like that but not blocking on it